### ISSUE TITLE
Throw during de-serialization

### ISSUE THREAD

**Issue by live1206** [aaubry/YamlDotNet#546]
Throw with 9.1.0 as below, works with 8.1.2:
```
IndexOutOfRangeException: Index was outside the bounds of the array.
    YamlDotNet.Core.InsertionQueue<T>.Enqueue(T)
    YamlDotNet.Core.Scanner.FetchValue()
    YamlDotNet.Core.Scanner.FetchNextToken()
    YamlDotNet.Core.Scanner.FetchMoreTokens()
    YamlDotNet.Core.Scanner.MoveNextWithoutConsuming()
    YamlDotNet.Core.Parser.GetCurrentToken()
    YamlDotNet.Core.Parser.ParseIndentlessSequenceEntry()
    YamlDotNet.Core.Parser.StateMachine()
    YamlDotNet.Core.Parser.MoveNext()
```

**Throwing implementation**:
```csharp
            var content = File.ReadAllText(filePath);
            var reader = new StringReader(content);
            new Deserializer().Deserialize(reader);
```

It even works  on 9.1.0 with the text reader as below, which does not read all the content
```csharp
            using var reader = File.OpenText(filePath);
            new Deserializer().Deserialize(reader);
```

**File content**:
```yml
# This file is automatically generated.
# Please do not modify this file manually, or your changes may be lost when
# documentation is rebuilt.
- items:
  - {name: AmpAmpByOracle, uid: Microsoft.Quantum.AmplitudeAmplification.AmpAmpByOracle}
  - {name: AmpAmpByOraclePhases, uid: Microsoft.Quantum.AmplitudeAmplification.AmpAmpByOraclePhases}
  - {name: AmpAmpByReflectionPhases, uid: Microsoft.Quantum.AmplitudeAmplification.AmpAmpByReflectionPhases}
  - {name: AmpAmpObliviousByOraclePhases, uid: Microsoft.Quantum.AmplitudeAmplification.AmpAmpObliviousByOraclePhases}
  - {name: AmpAmpObliviousByReflectionPhases, uid: Microsoft.Quantum.AmplitudeAmplification.AmpAmpObliviousByReflectionPhases}
  - {name: AmpAmpPhasesFixedPoint, uid: Microsoft.Quantum.AmplitudeAmplification.AmpAmpPhasesFixedPoint}
  - {name: AmpAmpPhasesStandard, uid: Microsoft.Quantum.AmplitudeAmplification.AmpAmpPhasesStandard}
  - {name: AmpAmpRUSByOracle, uid: Microsoft.Quantum.AmplitudeAmplification.AmpAmpRUSByOracle}
  - {name: AmpAmpRotationToReflectionPhases, uid: Microsoft.Quantum.AmplitudeAmplification.AmpAmpRotationToReflectionPhases}
  - {name: AmplitudeAmplificationFromPartialReflections, uid: Microsoft.Quantum.AmplitudeAmplification.AmplitudeAmplificationFromPartialReflections}
  - {name: AmplitudeAmplificationFromStatePreparation, uid: Microsoft.Quantum.AmplitudeAmplification.AmplitudeAmplificationFromStatePreparation}
  - {name: ApplyAmplitudeAmplification, uid: Microsoft.Quantum.AmplitudeAmplification.ApplyAmplitudeAmplification}
  - {name: ApplyFixedPointAmplification, uid: Microsoft.Quantum.AmplitudeAmplification.ApplyFixedPointAmplification}
  - {name: ApplyObliviousAmplitudeAmplification, uid: Microsoft.Quantum.AmplitudeAmplification.ApplyObliviousAmplitudeAmplification}
  - {name: ApplyTargetStateReflectionOracle, uid: Microsoft.Quantum.AmplitudeAmplification.ApplyTargetStateReflectionOracle}
  - {name: FixedPointReflectionPhases, uid: Microsoft.Quantum.AmplitudeAmplification.FixedPointReflectionPhases}
  - {name: ObliviousAmplitudeAmplificationFromPartialReflections, uid: Microsoft.Quantum.AmplitudeAmplification.ObliviousAmplitudeAmplificationFromPartialReflections}
  - {name: ObliviousAmplitudeAmplificationFromStatePreparation, uid: Microsoft.Quantum.AmplitudeAmplification.ObliviousAmplitudeAmplificationFromStatePreparation}
  - {name: ReflectionPhases, uid: Microsoft.Quantum.AmplitudeAmplification.ReflectionPhases}
  - {name: ReflectionStart, uid: Microsoft.Quantum.AmplitudeAmplification.ReflectionStart}
  - {name: RotationPhases, uid: Microsoft.Quantum.AmplitudeAmplification.RotationPhases}
  - {name: RotationPhasesAsReflectionPhases, uid: Microsoft.Quantum.AmplitudeAmplification.RotationPhasesAsReflectionPhases}
  - {name: StandardAmplitudeAmplification, uid: Microsoft.Quantum.AmplitudeAmplification.StandardAmplitudeAmplification}
  - {name: StandardReflectionPhases, uid: Microsoft.Quantum.AmplitudeAmplification.StandardReflectionPhases}
  - {name: TargetStateReflectionOracle, uid: Microsoft.Quantum.AmplitudeAmplification.TargetStateReflectionOracle}
  name: Microsoft.Quantum.AmplitudeAmplification
  uid: Microsoft.Quantum.AmplitudeAmplification
- items:
  - {name: AddConstantFxP, uid: Microsoft.Quantum.Arithmetic.AddConstantFxP}
  - {name: AddFxP, uid: Microsoft.Quantum.Arithmetic.AddFxP}
  - {name: AddI, uid: Microsoft.Quantum.Arithmetic.AddI}
```

**Comment by PJB3005** [comment]
I also ran into this exception with [this file](https://github.com/space-wizards/space-station-14/blob/4128e20c82eeaae802f145bb134aa920e1cc3499/Resources/Credits/Patrons.json) (yes I know it's JSON, YAML is a strict superset I promise I have a good reason)

**Comment by yufeih** [comment]
This issue still occur in 9.1.1

**Comment by samsmithnz** [comment]
I'm seeing this too. I can replicate it pretty easily. Downgrading to 8.1.2 resolves the issue.

**Comment by samsmithnz** [comment]
@yufeih, @PJB3005, & @live1206 I believe this is fixed with 9.1.4 too. My tests are now passing again.

cc @aaubry

**Comment by live1206** [comment]
Thanks @samsmithnz! Verified in 9.1.4

cc @aaubry
